test/unit/gateways/adyen_test.rb

```diff
@@ -810,6 +810,17 @@ def test_authorize_with_network_tokenization_credit_card
     assert_success response
   end
 
+  def test_authorize_with_sub_merchant_id
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card, @options.merge(sub_merchant_id: '12345abcde67890'))
+    end.check_request do |endpoint, data, headers|
+      parsed = JSON.parse(data)
+      puts parsed['additionalData']
```

Comment:
Might be good form to remove this `puts`. Otherwise lgtm!

Reply:
It's just one of those days

Reply:
That's why we have code review. :D